src/Http/Controllers/Admin/TranslationsController.php

```diff
@@ -67,4 +73,196 @@ public function update(UpdateTranslation $request, Translation $translation)
         return redirect('admin/translation');
     }
 
+    public function export(UpdateTranslation $request)
+    {
+        return Excel::download(new TranslationsExport($request), 'translations.xlsx');
+    }
+
+    public function import(ImportTranslation $request) // FIXME create separate request class
+    {
+        if ($request->hasFile('fileImport')){
+            // FIXME extract all this code with helper private methods
+            $chooseLanguage = strtolower($request->importLanguage);
+            // let's grab only the first sheet
+            $collection = (new TranslationsImport())->toCollection(request()->file('fileImport'))->first();
+
+            $existingTranslations = Translation::all()->filter(function($translation ) use ($chooseLanguage){
+                if(isset($translation->text->{$chooseLanguage})){
```

Comment:
I am not sure, that this is working as expected. I sense that this could be done in db query.